File: mysql/jdbc/src/TwoInsertRecords.java
import cubrid.jdbc.driver.CUBRIDConnection;
import cubrid.jdbc.driver.CUBRIDPreparedStatement;

import java.util.Date;

public class TwoInsertRecords {
	public static void main(String[] args) throws Exception {
		CUBRIDConnection conn = ShardConnection.getCUBRIDConnection();
		int totalRecordsToInsert = 5000;

		try {
			// post_id, title, content, post_date
			String sql = "INSERT INTO tbl_posts VALUES (/*+ shard_key */ ?, ?, ?, ?)";

			CUBRIDPreparedStatement preStmt = (CUBRIDPreparedStatement)conn.prepareStatement(sql);

			int startTime = (int)(new Date().getTime());

			for (int i = 1; i <= totalRecordsToInsert; ++i) {
				preStmt.setInt(1, i);
				preStmt.setString(2, "Post " + i);
				preStmt.setString(3, "Post " + i + " content");

				preStmt.setInt(4, (int) (new Date().getTime()/1000));

				preStmt.executeInsert();
			}

			int endTime = (int)(new Date().getTime());

			preStmt.close();
			conn.close();

			System.out.println("Connection is closed.");
			System.out.println(totalRecordsToInsert + " records were inserted in " + (endTime - startTime) + " ms.");
		} catch ( Exception e ) {
			System.err.println("SQLException : " + e.getMessage());
			System.err.println(e);
		} finally {
			if ( conn != null ) conn.close();
		}
	}
}

File: mysql/nodejs/TwoInsertRecords.js
var CUBRID = require('node-cubrid'),
  async = require('async'),
  Result2Array = CUBRID.Result2Array,
  ActionQueue = CUBRID.ActionQueue,
  dbConf = {
    host: '10.11.12.14',
    port: 45011,
    user: 'shard',
    password: 'shard123',
    database: 'sharddb'
  },
  client = CUBRID.createCUBRIDConnection(dbConf.host, dbConf.port, dbConf.user, dbConf.password, dbConf.database);

client.connect(function (err) {
	if (err) {
		console.log(err);
	}
	else{
		console.log('Connected');

		client.setEnforceOldQueryProtocol(true);

		var startTime = new Date().getTime(),
			totalRecordsToInsert = 5000,
			insertedCount = 0;

		function insert () {
			++insertedCount;
			// post_id, title, content, post_date
			var insertSQL = "INSERT INTO tbl_posts VALUES (/*+ shard_key */ " +
				insertedCount +
				", 'Post " + insertedCount + "'" +
				", 'Post " + insertedCount + " content'" +
				", " + new Date().getTime()/1000 +
				")";

			client.execute(insertSQL, function (err) {
				if (err) {
					console.log(insertSQL);
					console.log(err);
					client.close();
				}
				else{
					if (insertedCount == totalRecordsToInsert) {
						console.log(totalRecordsToInsert + " records were inserted in " + (new Date().getTime() - startTime) + " ms.");
						client.close();
					} else {
						insert();
					}
				}
			});
		}

		insert();
	}
});

client.on(client.EVENT_CONNECTION_CLOSED, function () {
  console.log('Connection closed.');
});
